schema/v1.1.0/dataset_config_models_extended.py

```diff
@@ -0,0 +1,321 @@
+
+
+@alru_cache
+async def is_go_id_ancestor(go_id_ancestor: str, go_id: str) -> bool:
+    url = f"https://api.geneontology.org/api/ontology/term/{go_id}/subgraph?start=0&rows=100"
+    headers = {"Accept": "application/json"}
+    async with aiohttp.ClientSession() as session, session.get(url, headers=headers) as response:
+        response_json = await response.json()
+        return go_id_ancestor in [ancestor["id"] for ancestor in response_json["ancestors"]]
+
+
+class ExtendedValidationAnnotationObject(AnnotationObject):
+    @model_validator(mode="after")
+    def validate_annotation_object(self) -> Self:
+        if not running_network_validation:
+            return self
+
+        # All of this checking is based on the id, so return if id is None
+        if self.id is None:
+            return self
+
+        # First ensure that the id is valid
+        valid_go_id, go_id_data = asyncio.run(validate_go_id(self.id))
+        if not valid_go_id:
+            raise ValueError(f"Invalid GO ID found in annotation object: {self.id}")
+
+        # Then check that the name matches
+        go_id_synonyms = go_id_data["synonyms"]
+
+        go_id_synonyms = [s.replace("@", "") for s in go_id_synonyms]
+        if self.name not in go_id_data["label"] and not any(
+            self.name in go_id_synonym for go_id_synonym in go_id_synonyms
+        ):
+            raise ValueError(f"Annotation object name '{self.name}' does not match id: {self.id}")
+
+        # Then check that the go_id is part of the cellular_component ontology
+        if not asyncio.run(is_go_id_ancestor(CELLULAR_COMPONENT_GO_ID, self.id)):
+            raise ValueError(f"Annotation object id {self.id} is not a cellular component")
+
+        return self
+
+
+# ==============================================================================
+# Annotation Publication Validation
+# ==============================================================================
+
+
+@alru_cache
+async def lookup_doi(doi: str) -> Tuple[str, bool]:
+    url = f"https://api.crossref.org/works/{doi}/agency"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return doi, response.status == 200
+
+
+@alru_cache
+async def lookup_empiar(empiar_id: str) -> Tuple[str, bool]:
+    url = f"https://www.ebi.ac.uk/empiar/api/entry/{empiar_id}/"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return empiar_id, response.status == 200
+
+
+@alru_cache
+async def lookup_emdb(emdb_id: str) -> Tuple[str, bool]:
+    url = f"https://www.ebi.ac.uk/emdb/api/entry/{emdb_id}"
+    async with aiohttp.ClientSession() as session, session.head(url) as response:
+        return emdb_id, response.status == 200
+
+
+PUBLICATION_REGEXES_AND_FUNCTIONS = {
+    "doi": (r"^(doi:)?10\.[0-9]{4,9}/[-._;()/:a-zA-Z0-9]+$", lookup_doi),
+    "empiar": (r"^EMPIAR-[0-9]{5}$", lookup_empiar),
+    "emdb": (r"^EMD-[0-9]{4,5}$", lookup_emdb),
+}
+
```

Comment:
^ is created for easier handling of the different types of publications and their according regexes / lookup functions

(if a list entry in a publication list matches a regex, then run the according lookup function to verify